storage/database/src/error.rs

```diff
@@ -7,6 +7,9 @@ use std::fmt::Debug;
 /// Alias for a thread-safe boxed error.
 type BoxError = Box<dyn std::error::Error + Send + Sync + 'static>;
 
+/// [`Result`] with [`RuntimeError`] as the error.
+pub type DbResult<T> = Result<T, RuntimeError>;
+
```

Comment:
`clippy::type_complexity` is more sensitive, there's some in `p2p/` silenced with `#[expect]`.

There's a pattern of doing `type Result<T> = Result<T, $common_crate_error>` [^1] [^2] [^3] so I thought we may as well fix https://github.com/Cuprate/cuprate/actions/runs/12071739897/job/33664158034?pr=353#step:11:51 with this type rather than `#[expect]`.

Lots of diffs in this PR are just moving over to `DbResult`. This could be `DatabaseResult` too but I think it's bit long for something that will appear in a lot of signatures. I think `Result` would clash / be confusing.

[^1]: https://doc.rust-lang.org/1.82.0/std/io/type.Result.html
[^2]: https://doc.rust-lang.org/1.82.0/std/fmt/type.Result.html
[^3]: https://docs.rs/heed/latest/heed/type.Result.html